wagtail/documents/views/serve.py

```diff
@@ -135,7 +135,7 @@ def authenticate_with_password(request, restriction_id):
 
     password_required_template = getattr(
         settings,
-        "DOCUMENT_PASSWORD_REQUIRED_TEMPLATE",
+        "WAGTAIL_DOCUMENT_PASSWORD_REQUIRED_TEMPLATE",
```

Comment:
We will need to have a backwards compatible lookup of the old value here. 

And if the old value is in use we need to raise a depreciation warning.